src/inc/quic_datapath.h

```diff
@@ -306,63 +361,89 @@ QuicDataPathResolveAddress(
     );
 
 //
-// The following APIs are specific to a single UDP port abstraction.
+// The following APIs are specific to a single UDP or TCP socket abstraction.
 //
 
 //
-// Creates a datapath binding handle for the given local address and/or remote
-// address. This function immediately registers for receive upcalls from the
-// UDP layer below.
+// Creates a UDP socket for the given local address and/or remote address. This
+// function immediately registers for receive upcalls from the layer below.
 //
 _IRQL_requires_max_(PASSIVE_LEVEL)
 QUIC_STATUS
-QuicDataPathBindingCreate(
+QuicSocketCreateUdp(
     _In_ QUIC_DATAPATH* Datapath,
     _In_opt_ const QUIC_ADDR* LocalAddress,
     _In_opt_ const QUIC_ADDR* RemoteAddress,
-    _In_opt_ void* RecvCallbackContext,
-    _Out_ QUIC_DATAPATH_BINDING** Binding
+    _In_opt_ void* CallbackContext,
+    _Out_ QUIC_SOCKET** Socket
+    );
+
+//
+// Creates a TCP socket for the (optional) (optional) given local address and
```

Comment:
(optional) (optional) is a bit weird here. Also doesn't match the UDP docs.

Reply:
Hmm, I think a copy/paste bug. I'll fix it up.